students/andrew_garcia/lesson_04/Mailroom Part 2.py

```diff
@@ -0,0 +1,167 @@
+'''
+Andrew Garcia
+Mailroom Part 2
+7/6/19
+'''
+
+#!/usr/bin/env python3
+
+all_donations = {'Jayson Black': [156.80, 207.32, 219.92],
+          'Bryan Fultan': [1236.28],
+          'Danica Dolores': [163.51, 100.42, 186.22],
+          'Skylar Odell': [167.90, 151.62],
+          'Roy Max': [137.97, 227.63]}
+
+def options_menu():
+    """Menu of options the user is shown, allowing them to navigate through the script """
+
+    answer = input('''
+Options:
+1 - Send a Thank You 
+2 - Send All Donors a Thank You
+3 - View a Report
+4 - Quit
+
+    Select an Option: ''')
+
+    if answer == '1':
+        send_thank_you()
+    elif answer == '2':
+        thank_all()
+    elif answer == '3':
+        create_report()
+    elif answer == '4':
+        pass
+    else:
+        print('That is not a valid answer!')
+        options_menu()
+
+
+def thank_all(all_donations = all_donations):
+    """Creates a text file for all donors and their donations"""
+
+    thank_donations = []  # create a list of donor and donation amount
+    donation_amount = []  # takes note of all donation amounts from one donor
+    for key in all_donations:
+        donation_amount += all_donations[key]
+        for item in donation_amount:
+            thank_donations += [[key, item]]
+        donation_amount = []
+    print(thank_donations)
+    for item in thank_donations:
+        filename = item[0] + '-' + str(item[1])  # creates file named after donor and donation amount
+        with open(f'{filename}.txt', 'w+') as f:
+            text = (f'''\nDear {item[0]},
+    
+            Thank you so much for your generous donation of ${item[1]:.2f}!
+            The kind generosity that you and others give us helps us complete the great work we are doing for others.
+    
+            From,
+            Your Local Charity
+            \n''')
+            f.write(text)
+    print('Thank You Saved.')
+    options_menu()
+
+def format_thank_you(name, amount):
+    """Creates thank you note for one donor"""
+
+    print(f'''\nDear {name.title()},
+
+Thank you so much for your generous donation of ${amount:.2f}!
+The kind generosity that you and others give us helps us complete the great work we are doing for others.
+
+From,
+Your Local Charity
+\n''')
+    options_menu()
+
+def existing_donor(name, all_donations = all_donations):
+    """Creates a Thank You Letter based on existing donations"""
+
+    if name.title() in all_donations:  # looks for donation amounts from a donor
```

Comment:
use dictionary efficiently by accessing your donor via a key to check if donor exists or not.

`if all_donations.get(name)` 
`if name in all_donations` searches for the name in a list of names instead.